ProcessLib/ComponentTransport/ComponentTransportFEM.h

```diff
@@ -131,48 +152,119 @@ class LocalAssemblerData : public ComponentTransportLocalAssemblerInterface
         auto local_b = MathLib::createZeroedVector<LocalVectorType>(
             local_b_data, local_matrix_size);
 
+        // Nodal DOFs include pressure
+        auto const num_comp = num_nodal_dof - 1;
+
+        // Get block matrices
+        auto Kpp = local_K.template block<pressure_size, pressure_size>(
+            pressure_index, pressure_index);
+        auto Mpp = local_M.template block<pressure_size, pressure_size>(
+            pressure_index, pressure_index);
+        auto Bp = local_b.template segment<pressure_size>(pressure_index);
+
+        auto local_p = Eigen::Map<const NodalVectorType>(
+            &local_x[pressure_index], pressure_size);
+
+        for (int comp_id = 0; comp_id < num_comp; ++comp_id)
```

Comment:
```suggestion
        for (int component_id = 0; component_id < number_of_components; ++component_id)
```
Also I suggest to keep variable's usage and definition close: move definition of the `number_of_components` before the for-loop.

Reply:
Fixed and stuck the condition variable 'number_of_component' with the follow-up for-loop.